client/src/document.js

```diff
@@ -100,69 +100,93 @@ function RenderSideBar({ doc }) {
     return null;
   }
   return doc.related_content.map(node => (
-    <SidebarLeaf
-      key={node.title}
-      depth={0}
-      title={node.title}
-      content={node.content || []}
-    />
+    <SidebarLeaf key={node.title} parent={node} />
   ));
 }
 
-function SidebarLeaf({ title, content }) {
-  const titleNode = <h3>{title}</h3>;
-  return (
-    <div>
-      {titleNode}
-      <ul>
-        {content.map(node => {
-          if (node.content) {
-            return (
-              <li key={node.title}>
-                <SidebarLeaflets node={node} />
-              </li>
-            );
-          } else {
-            return (
-              <li key={node.uri}>
-                <Link to={node.uri}>{node.title}</Link>
-              </li>
-            );
-          }
-        })}
-      </ul>
-    </div>
-  );
-}
-
-function SidebarLeaflets({ node }) {
+function SidebarLeaf({ parent }) {
+  const titleNode = <h3>{parent.title}</h3>;
```

Comment:
Why make a variable of this? It's not changing and it's not used repeatedly. I think you should just put the `<h3>{parent.title}</h3>` directly into line 135. And if that's not possible, if this is a hack due to mutations or something, then it needs to be why-explained with a code comment.